src/chains/ethereum/ethereum/tests/blockchain.test.ts

```diff
@@ -0,0 +1,114 @@
+
+      // set up wallet/blockchain
+      const wallet = new Wallet(options.wallet);
+      const initialAccounts = wallet.initialAccounts;
+      const blockchain = new Blockchain(options, initialAccounts[0].address);
+      await blockchain.initialize(wallet.initialAccounts);
+      const common = blockchain.common;
+
+      // set up two transactions
+      const [from, to] = wallet.addresses;
+      const secretKey = wallet.unlockedAccounts.get(from);
+      const rpcTx: EIP1559FeeMarketRpcTransaction = {
+        from,
+        to,
+        gasLimit,
+        maxFeePerGas: "0xffffffff",
+        type: "0x2",
+        nonce: "0x0"
+      };
+      const transaction1 = TransactionFactory.fromRpc(rpcTx, common);
+      const transaction2 = TransactionFactory.fromRpc(
+        { ...rpcTx, nonce: "0x1" },
+        common
+      );
+
+      const transaction1Promise = blockchain.queueTransaction(
+        transaction1,
+        secretKey
+      );
+      const transaction2Promise = blockchain.queueTransaction(
+        transaction2,
+        secretKey
+      );
+      // we're in strict mode, so transactions are in the pool and not yet mined
+      const transactionHashes = await Promise.all([
+        transaction1Promise,
+        transaction2Promise
+      ]);
+
+      const timestamps: number[] = [];
+      let i = 0;
+      const startingBlockNumber =
+        blockchain.blocks.latest.header.number.toNumber();
+
+      // wait for our two transactions to be mined by waiting for the chain to
+      // emit the block events. verify that it's the correct transaction, and
+      // store the timestamp
+      const assertBlocks = new Promise((resolve, reject) => {
+        const off = blockchain.on("block", async (block: Block) => {
+          const transactions = block.getTransactions();
+          if (transactions.length === 0) return;
+          // a failed assertion inside of this promise doesn't fail the test,
+          // it just stops this promise early. so catch any failures, reject the
+          // promise, and use that rejection to fail the test
+          try {
+            const blockNumber = block.header.number.toString();
+            const expectedBlockNumber = startingBlockNumber + i + 1;
+            assert.equal(blockNumber, `0x${expectedBlockNumber.toString(16)}`);
+            assert.equal(transactions.length, 1);
+
+            const transaction = transactions[0];
+            assert.equal(
+              transaction.hash.toString(),
+              transactionHashes[i].toString()
+            );
+
+            timestamps.push(block.header.timestamp.toNumber());
+            i++;
+            if (i === 2) {
+              off();
+              resolve(true);
+            }
+          } catch (e) {
+            off();
+            reject(e);
+          }
+        });
+      });
+      const success = (await assertBlocks) as any;
+      assert.equal(success, true, success);
```

Comment:
It's like a mini haiku! This should probably be `assert(success, <some message about what we expected but didn't get here>)`?